homeassistant/components/hue.py

```diff
@@ -0,0 +1,245 @@
+DEFAULT_ALLOW_UNREACHABLE = False
+
+PHUE_CONFIG_FILE = 'phue.conf'
+
+CONF_ALLOW_IN_EMULATED_HUE = "allow_in_emulated_hue"
+DEFAULT_ALLOW_IN_EMULATED_HUE = True
+
+CONF_ALLOW_HUE_GROUPS = "allow_hue_groups"
+DEFAULT_ALLOW_HUE_GROUPS = True
+
+BRIDGE_CONFIG_SCHEMA = vol.Schema([{
+    vol.Optional(CONF_HOST): cv.string,
+    vol.Optional(CONF_FILENAME): cv.string,
+    vol.Optional(CONF_ALLOW_UNREACHABLE): cv.boolean,
+    vol.Optional(CONF_ALLOW_IN_EMULATED_HUE): cv.boolean,
+    vol.Optional(CONF_ALLOW_HUE_GROUPS,
+                 default=DEFAULT_ALLOW_HUE_GROUPS): cv.boolean,
+}])
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_BRIDGES, default=[]): BRIDGE_CONFIG_SCHEMA,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+ATTR_GROUP_NAME = "group_name"
+ATTR_SCENE_NAME = "scene_name"
+SCENE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_GROUP_NAME): cv.string,
+    vol.Required(ATTR_SCENE_NAME): cv.string,
+})
+
+CONFIG_INSTRUCTIONS = """
+Press the button on the bridge to register Philips Hue with Home Assistant.
+
+![Location of button on bridge](/static/images/config_philips_hue.jpg)
+"""
+
+
+def setup(hass, config):
+    """Set up the Hue platform."""
+    config = config.get(DOMAIN)
+    if config is None:
+        config = {}
+
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    discovery.listen(
+        hass,
+        SERVICE_HUE,
+        lambda service, discovery_info:
+        bridge_discovered(hass, service, discovery_info))
+
+    bridges = config.get(CONF_BRIDGES, [])
+    if len(bridges) == 0:
+        _LOGGER.info("No bridges found in configuration")
+        return True
+
+    for bridge in bridges:
+        filename = bridge.get(CONF_FILENAME, PHUE_CONFIG_FILE)
+        allow_unreachable = bridge.get(CONF_ALLOW_UNREACHABLE,
+                                       DEFAULT_ALLOW_UNREACHABLE)
+        allow_in_emulated_hue = bridge.get(CONF_ALLOW_IN_EMULATED_HUE,
+                                           DEFAULT_ALLOW_IN_EMULATED_HUE)
+        allow_hue_groups = bridge.get(CONF_ALLOW_HUE_GROUPS,
+                                      DEFAULT_ALLOW_HUE_GROUPS)
+
+        host = bridge.get(CONF_HOST, None)
+
+        if host is None:
+            host = _find_host_from_config(hass, filename)
+
+        if host is None:
+            _LOGGER.error("No host found in configuration")
+            return False
+
+        setup_bridge(host, hass, filename, allow_unreachable,
+                     allow_in_emulated_hue, allow_hue_groups)
+
+    discovery.load_platform(hass, 'light', DOMAIN, {}, config)
+
+    return True
+
+
+def bridge_discovered(hass, service, discovery_info):
+    """Dispatcher for Hue discovery events."""
+    host = discovery_info.get('host')
+    serial = discovery_info.get('serial')
+
+    filename = 'phue-{}.conf'.format(serial)
+    setup_bridge(host, hass, filename)
+
+    discovery.load_platform(hass, 'light', DOMAIN, discovery_info, {})
```

Comment:
We should fire this inside `HueBridge.setup()` and pass the hue bridge ID as discovery info. The light platform can then pick the right bridge and just expose those entities.

Light platform should also make sure to cancel setup when `discovery_info is None`, as a discovery event is always fired when a bridge is connected.